src/__tests__/rules.test.ts

```diff
@@ -76,4 +82,54 @@ describe('rules', () => {
       expect(() => require(`../rules/${ruleName}`)).not.toThrow();
     });
   });
+
+  describe('rule documentation files have the correct content', () => {
+    it.each(ruleNames)('%s', ruleName => {
+      const rule = plugin.rules[ruleName];
+      const documentPath = join('docs', 'rules', `${ruleName}.md`);
+      const documentContents = readFileSync(documentPath, 'utf8');
+      const documentLines = documentContents.split(EOL);
+
+      // Check title.
+      const expectedTitle = `# ${rule.meta.docs.description} (\`${ruleName}\`)`;
+
+      expect(documentLines[0]).toStrictEqual(expectedTitle);
+
+      // Ensure that expected notices are present in the correct order.
+      const noticeLines = getRuleNoticeLines(ruleName);
+      const NOTICE_START_LINE = 3;
+
+      noticeLines.forEach((noticeLine, index) =>
+        expect(documentLines[index + NOTICE_START_LINE]).toStrictEqual(
+          noticeLine,
+        ),
+      );
+
+      // Ensure that unexpected notices are not present.
+      const { unexpectedNotices } = getNoticesForRule(rule);
+
+      unexpectedNotices.forEach(unexpectedNotice => {
+        expect(
+          documentContents.includes(MESSAGES[unexpectedNotice]),
+        ).toStrictEqual(false);
+      });
+
+      // Check for Rule Details section.
+      expect(documentContents).toContain('## Rule Details');
+
+      // Check if the rule has configuration options.
+      if (
+        (Array.isArray(rule.meta.schema) && rule.meta.schema.length > 0) ||
+        (typeof rule.meta.schema === 'object' &&
+          Object.keys(rule.meta.schema).length > 0)
+      ) {
+        // Should have an options section header:
+        expect(documentContents).toContain('## Options');
+      } else {
+        // Should NOT have any options section header:
+        expect(documentContents.includes('## Options')).toStrictEqual(false);
```

Comment:
it doesn't matter since we'll be axing this test, but fwiw you should be able to use `not.toContain` here